// web/src/app/layout.tsx
import type { Metadata } from "next";
import { Geist, Geist_Mono, Inter, Playfair_Display, DM_Sans, Space_Grotesk } from "next/font/google";
import "./globals.css";
import { ThemeToggle } from "../components/ThemeToggle";

const geistSans = Geist({
  variable: "--font-geist-sans",
  subsets: ["latin"],
});

const geistMono = Geist_Mono({
  variable: "--font-geist-mono",
  subsets: ["latin"],
});

const inter = Inter({
  subsets: ["latin"],
  variable: "--font-base-default",
});

const playfair = Playfair_Display({
  subsets: ["latin"],
  variable: "--font-base-theme1",
});

const dmSans = DM_Sans({
  subsets: ["latin"],
  variable: "--font-base-theme2",
});

const spaceGrotesk = Space_Grotesk({
  subsets: ["latin"],
  variable: "--font-base-theme3",
});

export const metadata: Metadata = {
  title: "Create Next App",
  description: "Generated by create next app",
};

export default function RootLayout({
  children,
}: Readonly<{
  children: React.ReactNode;
}>) {
  return (
    <html
      lang="en"
      className={`${geistSans.variable} ${geistMono.variable} ${inter.variable} ${playfair.variable} ${dmSans.variable} ${spaceGrotesk.variable} h-full antialiased`}
      suppressHydrationWarning
    >
      <head>
        <script dangerouslySetInnerHTML={{ __html: `(function(){var t=localStorage.getItem('fmds-theme');if(t&&['theme1','theme2','theme3'].includes(t))document.documentElement.setAttribute('data-theme',t);else if(t)localStorage.removeItem('fmds-theme');})();` }} />
      </head>
      <body className="min-h-full flex flex-col">
        <ThemeToggle />
        {children}
      </body>
    </html>
  );
}
